OpticalCommunications-2023-N-robots-TCP/python_implementation/threads/Logger.py

```diff
@@ -0,0 +1,383 @@
+import sqlite3
+import datetime
+import csv
+import logging
+import psutil
+import os
+import time
+from jtop import jtop
+
+
+class Logger:
+    DEFAULT_DB = 'logger.db'
+    
+
+    def __init__(self, logFilePath = None):
+        self.__jetson = jtop()
+        self.__jetson.start()
+
+            
+        if logFilePath == None:
+           logFilePath = self.DEFAULT_DB
+
+
+        self.conn = sqlite3.connect(logFilePath)    
+        self.cursor = self.conn.cursor()
+
+        self.conn.commit() 
+
+        try:
+            self.cursor.execute ('''SELECT Module FROM eventTable ''')
+        except:
+            self.cursor.execute(
+                """ CREATE TABLE IF NOT EXISTS eventTable (
+                    ID INTEGER PRIMARY KEY,
+                    Timestamp TIMESTAMP(6) DEFAULT CURRENT_TIMESTAMP,
+                    ProcessID INTEGER,
+                    Tag VARCHAR(10), 
+                    Module  VARCHAR(10),
+                    LevelNum INTEGER,
+                    Message VARCHAR(255)
+                )"""
```

Comment:
There should be a semi-colon after the parenthesis and before """ to denote the end of the execution string.
Also I tested this on the nano and it did not recreate the table each time so the try-exceptions can be removed to simplify the code and remove ambiguity. This should be done after merging in though, merge in first and keep this as task for after.